Add max-depth flag and set|unset_max_depth internals

mcky opened the pull request:
As discussed in #843 this adds the flag `--max-depth=N` and the internals/verbs `:set_max_depth N` and `:unset_max_depth`

Demo video of the new functionality:

https://github.com/user-attachments/assets/031685d6-33ef-458b-8d27-71874ea792ee

I'm a rust novice and I'm not precious about this code at all, if I've gone in the wrong direction I'm happy to hop on miaou and discuss.

**Notes**
- The depth passed doesn't include the current folder, which I think is more intuitive
- `--max-depth=0` is allowed but basically only shows the current directory, so is useless. If we want to validate this, I wasn't sure where to do so
- I couldn't find any similar test suites to extend, so have only manually tested it

**Questions**
- Should I add this to `Conf`? 
- What should the short flag be? `-d` is taken and `-m` doesn't feel ideal
- Shall I add a verb shortcut? `md` / `umd`? 

As an aside I noticed there's a bit of mixed indentation in `panel_state.rs` and running `cargo fmt` creates a lot of changes (that'll teach me to run it without staging my changes..), so I've tried to format it as consistently as I can

mcky commented on src/app/panel_state.rs:
```diff
@@ -482,6 +482,38 @@ pub trait PanelState {
+                        &|o| {
+                            if let Ok(max_depth) = flags.parse::<usize>() {
+                                o.max_depth = Some(max_depth);
+                                "*max depth updated*"
```
I'd wanted to show the updated depth here but `with_new_options` takes `&'static str`. I could update it to `Cow` but I didn't want to meddle if there was a reason for that choice

Canop replied:
The reason was just that there was no need for a dynamic string until now.

It could become either a Cow or a String. As there are a lot of places to update, this can be done in a separate PR if you prefer.

mcky commented on src/tree_build/builder.rs:
```diff
@@ -216,6 +216,9 @@ impl<'c> TreeBuilder<'c> {
                 return None;
             }
         }
+        if self.options.max_depth.map_or(false, |max| depth as usize > max) {
+            return None;
+        }
```
Also not entirely sure if this is the right place to perform the check, I'd also tried adding it to `gather_lines` by the `!deep` check, but settled on here

Canop replied:
The test should be done in gather_lines, at this point: https://github.com/Canop/broot/blob/main/src/tree_build/builder.rs#L359

Canop replied:
The casting to usize is OK. the flag should probably be directly a `u16` but it doesn't really matter.

Canop commented:
> Should I add this to Conf?

Not for now.

> What should the short flag be? -d is taken and -m doesn't feel ideal

Let's keep it without short flag.

> Shall I add a verb shortcut? md / umd?

`md` is `make directory` right now. We can keep it without verb shortcut, as user can add it if desired.

mcky commented:
@Canop Thanks for the feedback and patience, I've moved the check to `gather_lines`, swapped for a u16 and dropped the shorthand flag. Left the Cow change as yeah I'm not confident in trying to quickly swap it out here

Canop commented:
It seems to work very well. I'll merge that.
